providers/whispercpp.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import json
+import os
+import subprocess
+import time
+from api.enums import WingmanInitializationErrorType
+from api.interface import WhispercppSttConfig, WhispercppTranscript, WingmanInitializationError
+from os import path
+
+class Whispercpp:
+    def __init__(self, wingman_name: str, base_url: str, autostart: bool, whispercpp_exe_path: str, whispercpp_model_path: str, temperature: float, language: str, times_checked_whispercpp: int):
+        self.wingman_name = wingman_name if wingman_name else ""
+        self.base_url = base_url if base_url else "http://127.0.0.1:8080"
+        self.autostart = autostart if autostart else False
+        self.whispercpp_exe_path = whispercpp_exe_path if whispercpp_exe_path else None
+        self.whispercpp_model_path = whispercpp_model_path if whispercpp_model_path else None
+        self.temperature = temperature if temperature else 0.0
+        self.language = language if language else "en"
+        self.times_checked_whispercpp = 0
+        
+    def validate_config(self, config: WhispercppSttConfig, errors: list[WingmanInitializationError]):
+        if not errors:
+            errors = []
+
+        if not config.base_url:
+            errors.append(
+                WingmanInitializationError(
+                    wingman_name=self.wingman_name,
+                    message="Missing base url for whispercpp. Please provide a url where whispercpp can be reached.  By default this is http://127.0.0.1:8080.",
+                    error_type=WingmanInitializationErrorType.INVALID_CONFIG,
+                )
+            )
+
+        if config.autostart == True and (config.whispercpp_exe_path == None or config.whispercpp_model_path == None or not os.path.exists(config.whispercpp_exe_path) or not os.path.exists(config.whispercpp_model_path)):
+            errors.append(
+                WingmanInitializationError(
+                    wingman_name=self.wingman_name,
+                    message="You chose to use whispercpp and chose autostart but did not provide proper paths to the whispercpp exe and model.  Please turn autostart off or set / confirm the paths to the exe and model.",
+                    error_type=WingmanInitializationErrorType.INVALID_CONFIG,
+                )
+            )
+
+        self.base_url = config.base_url if config.base_url else "http://127.0.0.1:8080"
```

Comment:
So my suggested change here is basically: remove all the `Optional`s in your config model(s - same for XVASynth), unless the providers can actually work without a config property. A good example for that is OpenAI `base_path`- you CAN provide this but don't have to and OpenAI will still work. I can also pass `url_base=None` to the OpenAI API and that's fine.

Doing this will remove the need for the ternaries here because if a model prop is not optional, Pydantic auto-validates for you that it's set in the config or will throw a `ValidationError` otherwise. 
Your specific `validate` function here then only does logical validation on top.

Reply:
I think I have it now, made some changes and going to test.